src/ai/defaultai.cc

```diff
@@ -2841,6 +2884,14 @@ bool DefaultAI::construct_building(const Time& gametime) {
 		assert(player_);
 		int32_t const maxsize = player_->get_buildcaps(bf->coords) & Widelands::BUILDCAPS_SIZEMASK;
 
+		// Some buildings needs to consider distance from nearest warehouse
+		// It is non-negative value, and should be deducted from prio for some productionsites
+		const int32_t wh_distance_malus = management_data.neuron_pool[31].get_result_safe(
```

Comment:
I think the calculated malus is not related to the actual value of the distance as it is simply the addition of 2 neuron weights at a neuron position determined by the flag distance. 
it would be better to claculate the malus directly from flag distance and scale it by magic number.

Reply:
What this neuron do is converts values <-20,20> to <-100,100>. Or if absvalue, then to <100,0>

And problem with magic numbers is that we are running out of them. And increasing them would mean compatibility break.

Reply:
sorry, it returns <-X,X> where X is up to 100

Reply:
Point of this that the relation input:output is not linear, but is logarithmic, parabolic or linear

Reply:
ok got it. in this case the second paret of the addition is maybe zero most of the time even if we have big distances (i.e. 50). so maybe better choose a divisor smaller then 25 (maybe 4 or 5) to get reasonable numbers for input.

Reply:
It is malus, so bigger distance => bigger malus. Up to that upper limit X 
Those dividers define where distance is cropped. I was thinking about tweaking it a bit. First one can be 1 (basically distance above 20 is cropped to 20) and second to be cropped to 60 (divider 3)

Reply:
I would not crop the first one at 20 as this is stil rather near a warehouse and would deliver the highst malus already so using 2 and 3 might be good which would increase malus with distance but has a bigger range

Reply:
OK, first one no divider and second one divider 3 - cropping at 60 - this is fine for me